enhancements/insights/insights-runtime-extractor.md

```diff
@@ -0,0 +1,643 @@
+* “Fingerprints” which are self-contained executables that run in the container’s process namespaces to identify what the container is running.
+
+
+
+
+The `exporter` component runs an unprivileged HTTP server. When it receives a `GET /gather-runtime-info`, it triggers an extraction by connecting to the `extractor` that replies
+with a path to a dir on the shared volume. The exporter then collects the raw data from that directory, bundle them in a JSON payload sent as the HTTP response.
+Connections to the `exporter` HTTP server will be fronted by `kube-rbac-proxy` so that only the Insights Operator is allowed to query it.
+
+The `exporter` image would contain:
+
+* The HTTP server (written in Go)
+
+Due to the self-contained nature of these executables, the 2 images that compose the insights-runtime-extractor are built from `scratch` (and not from an existing base image).
+
+### Risks and Mitigations
+
+#### Security Risk & Mitigation
+
+##### Hardened and Minimized Privileged Code
+
+Introspecting the runtime environment of an arbitrary running container requires certain root-like system capabilities. First, the ability to introspect the host process table (`/proc` on a root pid namespace) is necessary to detect key processes running in all containers. Second, the ability to switch to an arbitrary mount namespace hosting the inspected container’s overlay filesystem is required to be able to examine and identify file contents within the container.  Third, read & write access to the container runtime socket (`/var/run/crio.sock`) is needed to portably correlate container metadata with a discovered process. Finally, for safety reasons described below, the ability to switch to an arbitrary user is required to drop privileges yet retain the necessary permission to analyze file contents.
+
+Any code running under such a high level of privilege is an attractive target for privilege escalation, since it can be leveraged to gain complete access to the node’s underlying operating system, and all hosted data. It is therefore essential to minimize the potential attack surface area and harden all essential code of the insights-runtime-extractor component.
+
+##### Reducing Attack Surface Area through Modular Process Design
+
+Splitting the inspection logic into multiple plugins/modules not only allows for extensibility but also provides for the ability to partition critical privilege requiring code from code which requires only point in time access to a specific container. Following this approach, privileged actions can be limited to a thin coordinator process, with the bulk of logic executed as a forked downgraded process limited to the same access afforded the inspected container target. Additionally, limiting the lifespan of the downgraded logic’s process to a per-container execution basis mitigates potential attacks where a rogue container corrupts the inspection logic to modify or gain access to a second victim container. 
```

Comment:
Nit: This sections refer to `coordinator` process, while the word is not mentioned in any of the text above. Would be good to keep the terminology consistent to reduce confusion.